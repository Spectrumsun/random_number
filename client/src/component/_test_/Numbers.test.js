import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import moxios from 'moxios';
import Number from '../Numbers';

Enzyme.configure({ adapter: new Adapter() });
describe('<RandomNumber />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Number />);
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display saved numbers', async () => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
            success: true,
            message: "Number List",
            savedNumbers: ["0652653715", "0171453496", "0366597054"]
        }
      })
    })
    const component = mount(<Number />);
    const spy = jest.spyOn(component.instance(), 'getNumbers');
    await spy()
    const state = component.instance().state;
    expect(spy).toHaveBeenCalled();
    expect(state.numbers).toEqual( ["0652653715", "0171453496", "0366597054"]);
  });

  it('should generate new numbers', async () => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
            success: true,
            message: "Number List",
            newNumbers: ["0652653715", "0171453496", "0366597054"]
        }
      })
    })
    const component = mount(<Number />);
    const spy = jest.spyOn(component.instance(), 'handleGenerate');
    await spy()
    const state = component.instance().state;
    expect(spy).toHaveBeenCalled();
    expect(state.random).toEqual( ["0652653715", "0171453496", "0366597054"]);
  });


  it('should show the max number on click', () => {
    const component = mount(<Number />);
    const spy = jest.spyOn(component.instance(), 'sortAscending');
    component.setState({
      numbers:  ["0652653715", "0171453496", "0366597054"]
    })
    component.find('.space_button').simulate('click');
    const state = component.instance().state;
    expect(spy).toHaveBeenCalled();
    expect(state.numbers).toEqual([ '0171453496', '0366597054', '0652653715'])
  });

  it('should show the min number on click', () => {
    const component = mount(<Number />);
    const spy = jest.spyOn(component.instance(), 'sortDescending');
    component.setState({
      numbers:  ["0652653715", "0171453496", "0366597054"]
    })
    component.find('#sortDescending').simulate('click');
    const state = component.instance().state;
    expect(spy).toHaveBeenCalled();
    expect(state.numbers).toEqual(['0652653715', '0366597054', '0171453496'])
  });
});