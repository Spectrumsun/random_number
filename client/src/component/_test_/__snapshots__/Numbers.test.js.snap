// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RandomNumber /> renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse container"
      id="navbarNav"
    >
      <a
        className="navbar-brand "
        href="/"
      >
        Random Numbers
      </a>
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link active icon"
            id="getButton"
            onClick={[Function]}
          >
            View saved numbers
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <br />
  <div
    className="container jumbotron"
  >
    <h1
      className="display-8"
    >
      Random Phone Number Generator
    </h1>
    <button
      className="btn btn-primary btn-lg"
      id="handleGenerate"
      onClick={[Function]}
      role="button"
    >
      Generate New Numbers
    </button>
    <br />
  </div>
  <div
    className="container"
  >
    <div>
      <ul />
      <ul />
    </div>
  </div>
</div>
`;
